src/webgpu/api/validation/capability_checks/limits/minStorageBufferOffsetAlignment.spec.ts

```diff
@@ -0,0 +1,184 @@
+import { GPUConst } from '../../../../constants.js';
+import { isPowerOfTwo } from '../../../../util/math.js';
+
+import {
+  kMinimumLimitBaseParams,
+  makeLimitTestGroup,
+  MinimumLimitValueTest,
+  MinimumTestValue,
+} from './limit_utils.js';
+
+function getDeviceLimitToRequest(
+  limitValueTest: MinimumLimitValueTest,
+  defaultLimit: number,
+  minimumLimit: number
+) {
+  switch (limitValueTest) {
+    case 'atDefault':
+      return defaultLimit;
+    case 'overDefault':
+      return 2 ** (Math.log2(defaultLimit) + 1);
+    case 'betweenDefaultAndMinimum':
+      return 2 ** (((Math.log2(defaultLimit) + Math.log2(minimumLimit)) / 2) | 0);
+    case 'atMinimum':
+      return minimumLimit;
+    case 'underMinimum':
+      return 2 ** (Math.log2(minimumLimit) - 1);
+  }
+}
+
+function getTestValue(testValueName: MinimumTestValue, requestedLimit: number) {
+  switch (testValueName) {
+    case 'atLimit':
+      return requestedLimit;
+    case 'underLimit':
+      return 2 ** (Math.log2(requestedLimit) - 1);
+  }
+}
+
+function getDeviceLimitToRequestAndValueToTest(
+  limitValueTest: MinimumLimitValueTest,
+  testValueName: MinimumTestValue,
+  defaultLimit: number,
+  maximumLimit: number
+) {
+  const requestedLimit = getDeviceLimitToRequest(limitValueTest, defaultLimit, maximumLimit);
+  return {
+    requestedLimit,
+    testValue: getTestValue(testValueName, requestedLimit),
+  };
+}
+
+const limit = 'minStorageBufferOffsetAlignment';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBindGroup,at_over')
+  .desc(`Test using createBindGroup at and over ${limit} limit`)
+  .params(kMinimumLimitBaseParams)
+  .fn(async t => {
+    const { limitTest, testValueName } = t.params;
+    // note: LimitTest.maximum is the adapter.limits[limit] value
+    const { defaultLimit, maximumLimit: minimumLimit } = t;
+    const { requestedLimit, testValue } = getDeviceLimitToRequestAndValueToTest(
+      limitTest,
+      testValueName,
+      defaultLimit,
+      minimumLimit
+    );
+
+    await t.testDeviceWithSpecificLimits(
+      requestedLimit,
+      testValue,
+      async ({ device, testValue, shouldError }) => {
+        const storageBuffer = t.trackForCleanup(
+          device.createBuffer({
+            size: testValue * 2,
+            usage: GPUBufferUsage.STORAGE,
+          })
+        );
+
+        const module = device.createShaderModule({
+          code: `
+            @group(0) @binding(0) var<storage> v: f32;
+            @compute @workgroup_size(1) fn main() {
+              _ = v;
+            }
+          `,
+        });
+
+        const pipeline = t.createPipeline('createComputePipeline', module);
```

Comment:
nit: we don't need a pipeline to create the bind group layout, it could be created directly.

Reply:
done